packages/data-model/src/data-registration.ts

```diff
@@ -0,0 +1,18 @@
+import { Store, DataFactory } from 'n3'
+
+export class DataRegistration {
+  dataset: Store //FIXME DatasetCore n3 needs to update their types
+  registeredShapeTree: string
+
+  constructor(dataset: Store) { //FIXME DatasetCore n3 needs to update their types
+    this.dataset = dataset
+    const matches =  this.dataset.match(
+      null,
+      DataFactory.namedNode('http://www.w3.org/ns/solid/interop#registeredShapeTree'),
+      null,
+      null
+      )
+    //@ts-ignore FIXME n3 needs to update their types
+    this.registeredShapeTree = [...matches][0].object.value
```

Comment:
Use `.shift()` together with the nullable operator `?`  instead of direct index access of the first item in the array.

Reply:
I have another branch where I moved helper functions from *client* package to new one *utils*. I'm using that function for matching here so this will go away in next PR.